### ISSUE TITLE
Locale set in setLocale is only evaluated if the route is not defined

### ISSUE THREAD

**Issue by samgavinio** [zendesk/zendesk_api_client_php#373]
The `Locale::setLocale` and `Locale::getRoute` methods are supposed to be able to set the locale for HelpCenter API calls, however 

* setLocale relies on setting the [$resourceName](https://github.com/zendesk/zendesk_api_client_php/blob/1cfedfc38ce8f1fded1d85e023e11ba8ceba0e2f/src/Zendesk/API/Traits/Resource/Locales.php#L54) when `Locale::getRoute` is called which, with how the library is currently designed, is only re-evaluated if a `RouteException` is [thrown](https://github.com/zendesk/zendesk_api_client_php/blob/1cfedfc38ce8f1fded1d85e023e11ba8ceba0e2f/src/Zendesk/API/Resources/ResourceAbstract.php#https://github.com/zendesk/zendesk_api_client_php/blob/1cfedfc38ce8f1fded1d85e023e11ba8ceba0e2f/src/Zendesk/API/Resources/ResourceAbstract.php#L248-L250) and is handled by a resource trait such as [FindAll](https://github.com/zendesk/zendesk_api_client_php/blob/1cfedfc38ce8f1fded1d85e023e11ba8ceba0e2f/src/Zendesk/API/Traits/Resource/FindAll.php#L22-L29).

This results in `setLocale()` only really being usable if the route is not explicitly defined in a `setRoutes` call. 

Not only does setLocale have this limit in it's usability, but it also has a pretty hard to follow logic of having to rely on an exception to be thrown for it do what it's supposed to do.

**How to reproduce:**

```
$section = $client->helpCenter->categories($categoryId)->sections()->setLocale('fil')->create([
        'name' => 'A section in the locale FIL',
        'description' => 'A section in the locale FIL',
])
```

**What is expected**

This should make a call to

```
POST /api/v2/help_center/fil/categories/{$categoryId}/sections.json
```

**What is the actual result**

It instead ignores the locale and makes the call to 

```
POST /api/v2/help_center/categories/{$categoryId}/sections.json
```

**Comment by ecoologic** [comment]
Hi,

 thank you for your request,
are you still interested in a solution for this issue?

**Comment by ecoologic** [comment]
Closing due to inactivity. Re-open if needed.
